plugins/calibre.koplugin/metadata.lua

```diff
@@ -0,0 +1,246 @@
+--[[
+    This module implements functions for loading, saving and editing calibre metadata files.
+
+    Calibre uses JSON to store metadata on device after each wired transfer.
+    In wireless transfers calibre sends the same metadata to the client, which is in charge
+    of storing it.
+--]]
+
+local rapidjson = require("rapidjson")
+local logger = require("logger")
+local util = require("util")
+
+local unused_metadata = {
+    "application_id",
+    "author_link_map",
+    "author_sort",
+    "author_sort_map",
+    "book_producer",
+    "comments",
+    "cover",
+    "db_id",
+    "identifiers",
+    "languages",
+    "pubdate",
+    "publication_type",
+    "publisher",
+    "rating",
+    "rights",
+    "thumbnail",
+    "timestamp",
+    "title_sort",
+    "user_categories",
+    "user_metadata",
+    "_series_sort_",
+}
```

Comment:
> Please cancel to dismiss. Do not press Ok unless you know what you're doing (it will strip most of the metadata from metadata.calibre. The metadata can be recovered if you connect calibre to your Kobo again, but you were warned smile )

I assume these are the metadata that will be stripped, if that's the case, does that mean I lose covers/thumbnails?

Reply:
Nope. Cover is just a path to a temporary cover made by calibre. Thumbnail is a base? encoded  image that is not used by KOReader.

All the metadata is sent by calibre during wireless transfers. We just discard most of them to have a lean json file. We do the same if we "initialize" a directory containing calibre metadata to use as the wireless inbox.

The warning is there not because we strip the metadata but because each calibre driver has its own set of rules and naming conventions, so if you set a library sent using Kindle device driver as the library used as wireless client some things won't match (like logical path to files).

On search we don't strip any metadata. We just read the whole json file and extract a cache of relevant info.

Reply:
Said metadata *should* be rebuilt by the Calibre driver the next time you connect over the usual method (i.e., USB), AFAIR.